File: 13-roman-to-integer/roman-to-integer.cpp
class Solution {
public:
  std::map<char, int> mp = {
        {'I', 1},
        {'V', 5},
        {'X', 10},
        {'L', 50},
        {'C', 100},
        {'D', 500},
        {'M', 1000}
    };
    int romanToInt(string s) {
        int sum =0;
        for(int i =0;i<s.size();i++){
            int no=mp[s[i]];
            
            if(i<s.size()-1 and mp[s[i+1]]>mp[s[i]]){
                no=mp[s[i+1]]-mp[s[i]];
                i++;
            }
            // cout<<no<<" ";
            sum+=no;
        }
        return sum;
    }
};

File: 13-roman-to-integer/roman-to-integer.ts
function romanToInt(s: string): number {
    let mp: Map<string, number> = new Map();
    mp.set("I", 1);
    mp.set("IV", 4);
    mp.set("V", 5);
    mp.set("IX", 9);
    mp.set("X", 10);
    mp.set("XL", 40);
    mp.set("L", 50);
    mp.set("XC", 90);
    mp.set("C", 100);
    mp.set("CD", 400);
    mp.set("D", 500);
    mp.set("CM", 900);
    mp.set("M", 1000);
     let ans:number=0;
    for(let i =0;i<s.length; ){
        let a:number=mp.get(s[i]+s[i+1]);
        let b:number=mp.get(s[i]);
        if(a) {ans+=a; i+=2;}
        else { ans+=b ;i++;}
        // console.log(ans,"-----");

    }return ans;
};